sites/svelte-5-preview/src/routes/docs/content/03-appendix/03-deprecations.md

```diff
@@ -40,6 +40,51 @@ These functions run indiscriminately when _anything_ changes. By using `$effect.
 
 Note that using `$effect` and `$effect.pre` will put you in [runes mode](/docs/runes) — be sure to update your props and state accordingly.
 
+## `createEventDispatcher`
+
+`createEventDispatcher` returns a function from which you can dispatch custom events. The usage is somewhat boilerplate-y, but it was encouraged in Svelte 4 due to consistency with how you listen to dom events (via `on:click` for example).
+
+Svelte 5 introduces [event attributes](/docs/event-handlers) which deprecate event directives (`onclick` instead of `on:click`), and as such we also encourage you to use callback properties for events instead:
+
+```diff
+<script>
+-	import { createEventDispatcher } from 'svelte';
+-	const dispatch = createEventDispatcher();
++	let { onGreet } = $props();
```

Comment:
should we use lowercase names so it feels more consistent with DOM events? or is the inconsistency a feature rather than a bug?

```suggestion
+	let { ongreet } = $props();
```

Reply:
I'd say feature - most people will likely separate the words somehow

Reply:
You know, thinking about this further, I think the idiomatic thing to do would be to pass in `greet` as a prop, rather than having a deeply weird distinction between DOM events and component 'events'. That also simplifies the code a fair bit — instead of refactoring the `greet` function defined here to call `onGreet` instead of `dispatch('greet')`, we can just remove it altogether.

The way to think of this is that you're not defining some behaviour that happens in response to a 'greet' event, you're _defining what greet means_

Reply:
I don't quite follow - are you suggesting to call it `let { greet } = $props()` (fine with me), or are you suggesting to remove the prop in this specific example (not fine, because this is about showing how to do events in Svelte 5)?

Reply:
the first